app/hooks/APIHooks.js

```diff
@@ -14,9 +14,11 @@ export const useEligibilitiesForCategory = categoryID => {
   const [eligibilities, setEligibilities] = useState(null);
 
   useEffect(() => {
-    dataService.get(`/api/eligibilities?category_id=${categoryID}`).then(response => {
-      setEligibilities(response.eligibilities);
-    });
+    if (categoryID !== null) {
```

Comment:
For a bit more information about this change, because this custom hook was originally being called conditionally in the `*DiscoveryForm` components, when I changed those components to unconditionally call this hook, I had to now guard against the case where `categoryID` is null. Same goes for the other custom hook lower in this file.

This is actually how I initially discovered the React Hooks violations, since I was working on porting this file to TypeScript, and I was looking at how these hooks were being called, when I realized that we were illegally conditionally calling the hooks.